pages/extensions/amp_example_preview/util/example_document.py

```diff
@@ -0,0 +1,95 @@
+import re
+
+from amp_component_versions import *
+
+PREVIEW_INLINE = 'inline'
+PREVIEW_NONE = 'none'
+
+PREVIEW_MODES = [
+  PREVIEW_INLINE,
+  PREVIEW_NONE,
+]
+
+
+class ExampleDocument(object):
+
+  def __init__(self, example_source, export_attributes, index):
+    """
+    :type example_source: str
+    :type export_attributes: dict
+    :type index: int
+    """
+    self.index = index
+
+    self.head = self.get_tag_content(example_source, 'head')
+    self.custom_style = None
+
+    style_match = self.find_tag(self.head, 'style', 'amp-custom')
+    if style_match:
+      self.custom_style = style_match.group(1)
+      self.head = self.head[:style_match.start(0)] + self.head[style_match.end(0):]
+
+    self.body = self.get_tag_content(example_source, 'body')
+    if self.body is None:
+      self.body = example_source
+
+    self.imports = get_components(export_attributes.get('imports'))
+    self.template = get_component(export_attributes.get('template'))
+
+    self.preview = export_attributes.get('preview', PREVIEW_NONE)
+    if self.preview == PREVIEW_INLINE and (self.head or self.custom_style)\
+      or self.preview not in PREVIEW_MODES:
+      # no inline preview supported if head elements are present
+      self.preview = PREVIEW_NONE
+
+    self.playground = 'playground' not in export_attributes or export_attributes['playground'].lower() != 'false'
+
+  def has_amp_script(self):
+    return self.has_head_tag('style', 'amp[\da-z]*-boilerplate')
```

Comment:
This was probably intended to check for `v0.js`?

Reply:
Ooops, good that you spotted this one! Of course you are right.